src/Exception/CouldNotMigrateException.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace RebelCode\Migrations\Exception;
+
+use Dhii\Util\String\StringableInterface as Stringable;
+use Exception as RootException;
+use RebelCode\Migrations\MigratorInterface;
+
+/**
+ * Concrete implementation of an exception thrown when a migration fails.
+ *
+ * @since [*next-version*]
+ */
+class CouldNotMigrateException extends AbstractMigratorException implements CouldNotMigrateExceptionInterface
+{
+    /**
+     * The migration version that failed.
+     *
+     * @since [*next-version*]
+     *
+     * @var int|string|Stringable|null
+     */
+    protected $migrationVersion;
+
+    /**
+     * Constructor.
+     *
+     * @since [*next-version*]
+     *
+     * @param string|Stringable|null     $message          The error message, if any.
+     * @param int|null                   $code             The error code, if any.
+     * @param RootException|null         $previous         The previous exception for chaining, if any.
+     * @param MigratorInterface|null     $migrator         The migrator instance, if any.
+     * @param int|string|Stringable|null $migrationVersion The migration version that failed, if any.
+     */
+    public function __construct(
```

Comment:
What about direction? Could that be useful?

Reply:
Hmm, probably. Or the source and destination versions.

Reply:
What do you think about `CouldNotMigrateExceptionInterface` having `getSourceVersion()` in addition to the already existing `getMigrationVersion()`? This is mostly a question about naming.

Reply:
Whatever you think is more suitable of those two.